packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -217,10 +217,8 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
                     : outputConfig.preset
                   : null;
 
-                if (preset) {
-                  if (preset.prepareDocuments) {
-                    outputSpecificDocuments = await preset.prepareDocuments(filename, outputSpecificDocuments);
-                  }
+                if (preset && preset.prepareDocuments) {
```

Comment:
this rule https://tortilla-hq.slack.com/archives/C03B470CJCT/p1673028464154629 will catch this case :)